test/capwap_loc_provider_SUITE.erl

```diff
@@ -48,44 +48,70 @@ all() ->
      test_reconfig,
      test_error].
 
+capwap_app(LocConfig) ->
+    {application,
+     capwap,
+     [{vsn, dummy},
+      {id, dummy},
+      {env, [{location_provider, #{}}]},
+      {mod, {capwap_dummy_app, LocConfig}}]}.
+
 init_per_suite(Config) ->
-    % Dummy app env value to enable the calls
-    application:set_env(capwap, location_provider, #{}),
+    logger:update_primary_config(#{level => all}),
+    logger:update_formatter_config(default, template, [time," ",pid," ",level," ",mfa,": ",msg,"\n"]),
+    logger:update_formatter_config(default, level, all),
+    logger:update_formatter_config(cth_log_redirect, template, [time," ",pid," ",level," ",mfa,": ",msg,"\n"]),
+    logger:update_formatter_config(cth_log_redirect, level, all),
+    % Dummy app to allow 
+    ct:pal("--> Apps: ~p", [application:which_applications()]),
+    ct:pal("--> My app: ~p", [application:get_application()]),
     ct:pal("Env for capwap: ~p", [application:get_all_env(capwap)]),
     ct:log("Suite config: ~p~n", [Config]),
     meck:new(test_loc_handler, [no_link, passthrough]),
-    logger:update_primary_config(#{level => all}),
-    application:ensure_all_started(ranch),
-    application:ensure_all_started(hackney),
     RefreshTime = 5000,
     LocConfig = #{providers => [
 	{capwap_loc_provider_http, #{uri => "http://127.0.0.1:9990", timeout => 30000}},
 	{capwap_loc_provider_default, #{default_loc => {location, <<"123.123">>, <<"456.456">>}}}
       ],
       refresh => RefreshTime},
+    %% NOTICE
+    %% Due to the dependency on checking the env of the application the caller
+    %% belongs to, this test suite sets up a dummy application called "capwap"
+    %% that includes both the location user and the location provider. Otherwise,
+    %% the caller function calls would fail when trying to find the appropriate
+    %% env config key (config woul be undefined since ct does not run within an
+    %% application).
+    %% So all calls on capwap_loc_provider are done on loc_provider_proxy instead,
+    %% which basically proxies the calls to capwap_loc_provider, but is also
+    %% under the capwap application, so the right env is chosen.
+    %% This is only needed for tests.
+    application:load(capwap_app(LocConfig)),
+    application:start(capwap),
+    application:ensure_all_started(ranch),
+    application:ensure_all_started(hackney),
```

Comment:
the `hackney` is already included into `capwap` - it no make sense to start it twice. Just call `application:ensure_all_started(capwap)` at the 89 line

Reply:
Because this is another capwap ^_^